server/item/recipe/item.go

```diff
@@ -0,0 +1,101 @@
+package recipe
+
+import (
+	"github.com/df-mc/dragonfly/server/item"
+	"github.com/df-mc/dragonfly/server/world"
+)
+
+// Item is a recipe item. It is an item stack inherently, but it also has an extra value for if it applies to all types.
+type Item struct {
```

Comment:
This name doesn't feel coherent with the existing item.Stack. Maybe rename this to `Input` or `InputItems`?